pkgs/by-name/pa/parabolic/package.nix

```
@@ -1,77 +1,82 @@
-, python3
-, ffmpeg
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  libnick,
+  boost,
+  curl,
+  glib,
+  libsecret,
+  gtk4,
+  libadwaita,
+  wrapGAppsHook4,
+  libxmlxx5,
+  blueprint-compiler,
+  yt-dlp,
+  ffmpeg,
+  aria2,
 }:
 
-buildDotnetModule rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "parabolic";
-  version = "2024.5.0";
+  version = "2024.11.1";
 
   src = fetchFromGitHub {
     owner = "NickvisionApps";
     repo = "Parabolic";
-    rev = version;
-    hash = "sha256-awbCn7W7RUSuEByXxLGrsmYjmxCrwywhhrMJq/iM1Uc=";
-    fetchSubmodules = true;
+    rev = "${finalAttrs.version}";
+    hash = "sha256-Vm5QVmPc0uipLlDy5Wn9S7HQ/0rLg0wVk0XdWRECQDY=";
   };
 
-  dotnet-sdk = dotnetCorePackages.sdk_8_0;
-  dotnet-runtime = dotnetCorePackages.runtime_8_0;
-  pythonEnv = python3.withPackages(ps: with ps; [ yt-dlp ]);
-
-  projectFile = "NickvisionTubeConverter.GNOME/NickvisionTubeConverter.GNOME.csproj";
-  nugetDeps = ./deps.nix;
-  executables = "NickvisionTubeConverter.GNOME";
-
-   nativeBuildInputs = [
+  nativeBuildInputs = [
+    cmake
     pkg-config
     wrapGAppsHook4
-    glib
-    shared-mime-info
-    gdk-pixbuf
     blueprint-compiler
   ];
 
-  buildInputs = [ gtk4 libadwaita ];
-
-  runtimeDeps = [
+  buildInputs = [
+    libnick
+    boost
+    curl
+    libsecret
+    glib
     gtk4
     libadwaita
-    glib
-    gdk-pixbuf
+    libxmlxx5
   ];
 
-  postPatch = ''
-    substituteInPlace NickvisionTubeConverter.Shared/Linux/org.nickvision.tubeconverter.desktop.in --replace '@EXEC@' "NickvisionTubeConverter.GNOME"
+  cmakeFlags = [
+    "-DUI_PLATFORM=gnome"
+  ];
+
+  runtimeDependencies = [
+    yt-dlp
+    ffmpeg
+    aria2
+  ];
+
+  dontWrapGApps = true;
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
   '';
 
   postInstall = ''
-    install -Dm444 NickvisionTubeConverter.Shared/Resources/org.nickvision.tubeconverter.svg -t $out/share/icons/hicolor/scalable/apps/
-    install -Dm444 NickvisionTubeConverter.Shared/Resources/org.nickvision.tubeconverter-symbolic.svg -t $out/share/icons/hicolor/symbolic/apps/
-    install -Dm444 NickvisionTubeConverter.Shared/Linux/org.nickvision.tubeconverter.desktop.in -T $out/share/applications/org.nickvision.tubeconverter.desktop
+    wrapProgram $out/bin/org.nickvision.tubeconverter \
+      --prefix PATH : ${lib.makeBinPath finalAttrs.runtimeDependencies}
   '';
 
-  makeWrapperArgs = [ "--prefix PATH : ${lib.makeBinPath [ pythonEnv ffmpeg ]}" ];
-
-  passthru.updateScript = ./update.sh;
-
   meta = with lib; {
-    description = "Download web video and audio";
+    description = "Graphical frontend for yt-dlp to download video and audio";
     homepage = "https://github.com/NickvisionApps/Parabolic";
-    license = licenses.mit;
-    maintainers = with maintainers; [ ewuuwe ];
-    mainProgram = "NickvisionTubeConverter.GNOME";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [
+      ewuuwe
+      normalcea
```

```suggestion
      normalcea
      getchoo
```
Feel free to add me as well